ui/app/AppLayouts/Chat/views/ChatMessagesView.qml

```diff
@@ -65,16 +81,28 @@ Item {
         }
 
         function onScrollToMessage(messageIndex) {
-            chatLogView.positionViewAtIndex(messageIndex, ListView.Center);
-            chatLogView.itemAtIndex(messageIndex).startMessageFoundAnimation();
+            chatLogView.positionViewAtIndex(messageIndex, ListView.Center)
+            chatLogView.itemAtIndex(messageIndex).startMessageFoundAnimation()
         }
 
-        // Not Refactored Yet
-        //            onNewMessagePushed: {
-        //                if (!chatLogView.scrollToBottom()) {
-        //                    newMessages++
-        //                }
-        //            }
+        function onMessageSearchOngoingChanged() {
+            d.markAllMessagesReadIfMostRecentMessageIsInViewport()
+        }
+    }
+
+    Connections {
+        target: d.chatDetails
+
+        function onActiveChanged() {
+            d.markAllMessagesReadIfMostRecentMessageIsInViewport()
+        }
+
+        function onHasUnreadMessagesChanged() {
```

Comment:
Do we really need this connection? Don't you already add the marker from the Nim side when new messages come in?

Reply:
See https://github.com/status-im/status-desktop/pull/8756#discussion_r1047195760